ntfs/filesystem/__init__.py

```diff
@@ -376,20 +407,29 @@ def __getitem__(self, index):
         if index < 0:
             index = len(self) + index
 
+        clusters = self._clusters
+        csize = clusters.get_cluster_size()
+
         # units: clusters
         for cluster_offset, num_clusters in self._runentries:
             # units: bytes
-            run_length = num_clusters * self._clusters.get_cluster_size()
+            run_length = num_clusters * csize
+            right_border = current_run_start_offset + run_length
 
-            if current_run_start_offset <= index < current_run_start_offset + run_length:
+            # Check if the target byte in the run entry
+            if current_run_start_offset <= index < right_border:
                 # units: bytes
-                i = index - current_run_start_offset
-                cluster = self._clusters[cluster_offset]
-                return cluster[i]
-
+                target_idx = index - current_run_start_offset
+                # The index of the cluster that contains the target byte
+                target_cluster_idx = int(target_idx/csize)
+                # The index of the target byte relative to the cluster
+                byte_relative_idx = (target_idx - target_cluster_idx * csize)
+                cluster = clusters[cluster_offset+target_cluster_idx]
+                return cluster[byte_relative_idx]
+            # else looking at next run entry
```

Comment:
the addition of comments here and above will help other programmers read and understand this code more clearly. thank you.